src/components/form/trash-list.vue

```diff
@@ -82,7 +90,13 @@ export default {
       // tell parent component (ProjectOverview) to refresh regular forms list
       // (by emitting event to that component's parent)
       this.$emit('restore');
-    }
+    },
+    toggleTrashExpansion(evt) {
+      const projProps = this.currentUser.preferences.projects[this.project.id];
+      if (evt.newState === 'closed') {
+        projProps.formTrashCollapsed = true;
+      } else if (projProps.formTrashCollapsed) delete projProps.formTrashCollapsed;
```

Comment:
<blockquote>

```js
this.currentUser.preferences.projects[this.project.id].formTrashCollapsed = (evt.newState === 'closed');
```
</blockquote>

I do like how that looks!

> every time you enter the page you'll get a PUT if the trash is open.

Isn't there only a request if there's a `toggle` event? If you enter the page, and the trash is already open, and you don't do anything, is there a `toggle` event in that case?

> Now, I've forgone putting logic in the Preferences that would avoid requests to the backend if the value hasn't changed.

Yeah, logic like that doesn't seem strictly necessary at this point. The UI should already make it so that you can't reselect your current preference. For example, if the project sort is set to `latest`, you can't set it again to `latest`: you have to select something else first. I think we should avoid something like a write-back cache at this point: it feels like more logic than we need for now.

Reply:
Ah I see, that's inconvenient that the event can be fired without user action.

> this is not (necessarily) a Vue thing.

That sounds right. Vue doesn't change how native events work.

> Currently I also think it best to have the workaround for what is a **local** problem (how `<details open>` behaves wrt firing events) in that function, rather than making the Preferences "smart?".

That makes a lot of sense to me. :+1: Trying to solve the problem in `UserPreferences` would make it easy to overlook that there is a local problem at all.

> So in fact none of the demo case use sites of the new preference persistence functionality look particularly nice, none are as simple as directly using a value from Preferences inside the Vue template 😭

We'll have such a use site one day, I'm sure! At least the `UserPreferences` object has been designed to make that a possibility.